### PR TITLE
move demo providers into namespace observability

### PR DESCRIPTION
**Pull request by zirain**
**Please provide a description of this PR:**

https://github.com/istio/istio.io/pull/14566#issuecomment-1921107203

### CODE REVIEW

**Comment by joaopgrassi** [comment]
Does this also need to change? https://github.com/istio/istio/blob/master/manifests/charts/default/files/profile-demo.yaml#L19

**Review by joaopgrassi — APPROVED** [review]

**Review by hanxiaop — COMMENTED** [review]
I think we had some discussions about not deploying these in `istio-system`, so I wrote the demo which deployed the otel collector in `otel-collector` namespace.

**Comment by zirain** [comment]
> I think we had some discussions about not deploying these in `istio-system`, so I wrote the demo which deployed the otel collector in `otel-collector` namespace.

https://github.com/istio/istio/pull/43041#discussion_r1090780553 is this one?

**Comment by hanxiaop** [comment]
> > I think we had some discussions about not deploying these in `istio-system`, so I wrote the demo which deployed the otel collector in `otel-collector` namespace.
> 
> [#43041 (comment)](https://github.com/istio/istio/pull/43041#discussion_r1090780553) is this one?

Yes

**Review by howardjohn — CHANGES_REQUESTED** [review]
I would prefer to change the `start-otel-collector-service.sh` rather than this. Its a bad practice to deploy things in istio-system and we should work to avoid where possible

**Comment by zirain** [comment]
> I would prefer to change the `start-otel-collector-service.sh` rather than this. Its a bad practice to deploy things in istio-system and we should work to avoid where possible

I agree with this, so I'd like move `skywalking` and `otel-collector` to another namespace, what about `monitoring`?

**Comment by joaopgrassi** [comment]
> I agree with this, so I'd like move skywalking and otel-collector to another namespace, what about monitoring?

Other options could be `telemetry` or `observability`.

**Comment by zirain** [comment]
> > I agree with this, so I'd like move skywalking and otel-collector to another namespace, what about monitoring?
> 
> Other options could be `telemetry` or `observability`.



> > I agree with this, so I'd like move skywalking and otel-collector to another namespace, what about monitoring?
> 
> Other options could be `telemetry` or `observability`.

SG

**Review by howardjohn — APPROVED** [review]

File: manifests/helm-profiles/demo.yaml [hunk 1]
```diff
@@ -7,16 +7,16 @@ meshConfig:
   extensionProviders:
     - name: otel
       envoyOtelAls:
-        service: opentelemetry-collector.istio-system.svc.cluster.local
+        service: opentelemetry-collector.observability.svc.cluster.local
         port: 4317
     - name: skywalking
       skywalking:
-        service: tracing.istio-system.svc.cluster.local
+        service: tracing.observability.svc.cluster.local
```
Comment (howardjohn) [inline manifests/helm-profiles/demo.yaml:14]: `samples/addons/jaeger.yaml`  hardcodes to istio-system. Is that an issue? I see this is 'skywalking' so maybe not

Comment (zirain) [reply manifests/helm-profiles/demo.yaml]: skywalking is hardcode to `istio-system` in [skywalking.yaml](https://github.com/istio/istio/blob/100cb541ab38f8a467226f8865f6f61e581a2a35/samples/addons/extras/skywalking.yaml#L16), reverted and will do it later.

**Pull request merged**